insights/parsers/tests/test_awx_manage.py

```diff
@@ -27,28 +27,42 @@
     load_entry_point('awx==3.6.4', 'console_scripts', 'awx-manage')()
 """.strip()
 
+AWX_MANAGE_LICENSE = """
+{"contact_email": "test@redhat.com", "company_name": "test Inc", "instance_count": 100, "license_date": 1655092799, "license_type": "enterprise", "subscription_name": "Red Hat Ansible Automation, Standard (100 Managed Nodes)", "sku": "MCT3691", "support_level": "Standard", "product_name": "Red Hat Ansible Automation Platform", "valid_key": true, "satellite": null, "pool_id": "2c92808179803e530179ea5989a157a4", "current_instances": 1, "available_instances": 100, "free_instances": 99, "time_remaining": 29885220, "trial": false, "grace_period_remaining": 32477220, "compliant": true, "date_warning": false, "date_expired": false}
+""".strip()
 
-def test_ansible_tower_license():
-    ret = AnsibleTowerLicenseType(context_wrap(NO_LICENSE))
-    assert ret.type == 'none'
-    ret = AnsibleTowerLicenseType(context_wrap(STD_LICENSE))
+
+def test_ansible_tower_license_type():
+    ret = AnsibleTowerLicenseType(context_wrap(AWX_MANAGE_LICENSE))
     assert ret.type == 'enterprise'
 
 
-def test_ansible_tower_license_ab():
+def test_ansible_tower_license_ab_type():
     with pytest.raises(SkipException):
         AnsibleTowerLicenseType(context_wrap(NG_COMMAND_0))
 
     with pytest.raises(ContentException):
         AnsibleTowerLicenseType(context_wrap(NG_COMMAND_1))
 
-    with pytest.raises(ParseException):
+    with pytest.raises(ValueError):
         AnsibleTowerLicenseType(context_wrap(NG_COMMAND_2))
 
+    with pytest.raises(ValueError):
+        AnsibleTowerLicenseType(context_wrap(NO_LICENSE))
+
+
+def test_ansible_tower_license():
+    ret = AnsibleTowerLicense(context_wrap(AWX_MANAGE_LICENSE))
+    assert ret.get("license_type") == 'enterprise'
+    assert ret.get("instance_count") == 100
+    assert ret.get("time_remaining") == 29885220
+    assert ret.get("contact_email") == "test@redhat.com"
```

Comment:
The result of this test item should be failed. it should be `assert ret.get("contact_email") is None` as per the above filters.

Please refer the following functions to test the filters

Reply:
@xiangce - Updated according the example, however, the filter still can not work, please have a check.

Reply:
Sorry for the inaccurate comment, the test of filters should be applied to the test of the datasources, to the Parser, the filter doesn't work, it will reflect all the contents from the input (which is as after filtered)

Please just add the test for the datasource itself.

Reply:
@xiangce - Updated, please have a check.